### PR TITLE
Disallow writing to reserved xt tables

### LINKED ISSUE #3245: Disallow DML against xt$txs
e.g. `INSERT INTO xt$txs (xt$id, foo) VALUES (10, 10)` (currently halts ingestion, prevents shutting down without memory leaks, user probably needs to start fresh etc.)

### PR DESCRIPTION
**Pull request by FiV0**
Mainly just what it says in the title. I don't know if there is some way to check this more efficiently.

### CODE REVIEW

**Comment by FiV0** [comment]
Do I actually need to check for explict `put/delete/erase` or should this happen client side?

File: core/src/main/clojure/xtdb/indexer.clj [hunk 1]
```diff
@@ -62,6 +62,13 @@
 
 (def ^:private abort-exn (err/runtime-err :abort-exn))
 
+(def ^:private forbidden-tables #{"xt/txs" "xt/tx_fns"})
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/indexer.clj]: could do with checking the schema for this one - I'd assume `xt.*` (_except_ xt/tx_fns, ironically - I would guess that this should break the tx-fn tests as new tx-fns are just inserted as put-docs), `information_schema.*` and `pg_catalog.*`?

File: src/test/clojure/xtdb/sql_test.clj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (ns xtdb.sql-test
   (:require [clojure.java.io :as io]
             [clojure.string :as str]
-            [clojure.test :as t]
+            [clojure.test :as t :refer [deftest]]
```
Comment (jarohen) [inline src/test/clojure/xtdb/sql_test.clj]: noo, I've only just gotten rid of that! :stuck_out_tongue:

**Comment by jarohen** [comment]
> Do I actually need to check for explict `put/delete/erase` or should this happen client side?

I'd check explicitly, yep - chances are the tx-log can be trusted in practice but no reason not to make sure. I might be tempted to check it submit-side too, for early feedback to the user.

> I don't know if there is some way to check this more efficiently.

wdym, sorry? efficiently as in code-size, it doesn't feel egregious to me, fwiw

File: src/test/clojure/xtdb/sql_test.clj (hunk 2 above)
Comment (FiV0) [reply src/test/clojure/xtdb/sql_test.clj]: Yeah, is there a reason for that? I always never refer but for `clojure.test` it seems idiomatic to me.

**Comment by FiV0** [comment]
>wdym, sorry? efficiently as in code-size, it doesn't feel egregious to me, fwiw

Checking every time we go through the indexing fns.

**Pull request merged**